src/main/java/org/jabref/gui/util/UiTaskExecutor.java

```diff
@@ -128,7 +128,7 @@ public <V> Future<?> schedule(BackgroundTask<V> task, long delay, TimeUnit unit)
     public void shutdown() {
         StateManager stateManager = Injector.instantiateModelOrService(StateManager.class);
         if (stateManager != null) {
-            stateManager.getBackgroundTasks().stream().filter(task -> !task.isDone()).forEach(Task::cancel);
+            stateManager.getRunningBackgroundTasks().stream().forEach(Task::cancel);
```

Comment:
The issue was that canceling a task removes items from the `runningBackgroundTasks` list since it's a filtered list. Removing items during iteration causes an `IndexOutOfBoundsException`.

Reverting this line will fix the issue.
I will reopen the PR.